test_scripts/Resumption/Handling_errors_from_HMI/010_unsubscribe_from_the_same_vehicledata_without_request_to_hmi_by_error_during_resumption.lua

```diff
@@ -0,0 +1,95 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0190-resumption-data-error-handling.md
+--
+-- Requirement summary:TBD
+--
+-- Description:
+-- In case:
+-- 1. SubscribeVehicleData_1, SubscribeVehicleData_2 are added by app1
+-- 2. SubscribeVehicleData_2, SubscribeVehicleData_3 are added by app2
+-- 3. Unexpected disconnect and reconnect are performed
+-- 4. App1 and app2 reregister with actual HashId
+-- 5. VehicleInfo.SubscribeVehicleData_2 requests for app2 and app1 are processed successful
+-- 6. VehicleInfo.SubscribeVehicleData_3 related to app1 is sent from SDL to HMI during resumption
+-- 7. HMI responds with error resultCode VehicleInfo.SubscribeVehicleData_3 request
+-- 8. HMI responds with success to remaining requests
+-- SDL does:
+-- 1. process unsuccess response from HMI
+-- 2. remove already restored data from app1, but does not send VehicleInfo.UnsubscribeVehicleData_2 to HMI
+-- 3. respond RegisterAppInterfaceResponse(success=true,result_code=RESUME_FAILED) to mobile application app1
+-- 4. restore all data for app2 and respond RegisterAppInterfaceResponse(success=true,result_code=SUCCESS)to mobile application app2
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/Resumption/Handling_errors_from_HMI/commonResumptionErrorHandling')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+-- [[ Local Variables ]]
+local vehicleDataSpeed = {
+  requestParams = { speed = true },
+  responseParams = { speed = { resultCode = "SUCCESS", dataType = "VEHICLEDATA_SPEED"} }
+}
+
+local vehicleDataRpm = {
+  requestParams = { rpm = true },
+  responseParams = { rpm = { resultCode = "SUCCESS", dataType = "VEHICLEDATA_RPM"} }
+}
+
+-- [[ Local Function ]]
+local function checkResumptionData()
+  common.getHMIConnection():ExpectRequest("VehicleInfo.SubscribeVehicleData")
+  :Do(function(_, data)
+      if data.params.speed then
+        local function sendResponse()
+          common.getHMIConnection():SendError(data.id, data.method, "GENERIC_ERROR", "info message")
+        end
+        RUN_AFTER(sendResponse, 300)
+      else
+        common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+      end
+    end)
+  :Times(4)
+
+  common.getHMIConnection():ExpectRequest("VehicleInfo.UnsubscribeVehicleData")
+  :Times(0)
```

Comment:
@GetmanetsIrina `VI.UnsubscribeVehicleData` shouldn't be sent for `gps` or `rpm`, but what about `speed`?

Reply:
Discussed. Added one more data item for app1.